web3/types.py

```diff
@@ -74,33 +77,53 @@ class AccessListEntry(TypedDict):
 
 AccessList = NewType("AccessList", Sequence[AccessListEntry])
 
+ABI_COMPONENT_DEPRECATION_MSG = "`{0}` is deprecated, use `{1}` instead."
+
 
-# todo: move these to eth_typing once web3 is type hinted
+@deprecated(
```

Comment:
Since this is going into `main`, I'd vote to remove these altogether, document in the `v7` migration notes, and deprecate in `v6`. That applies to all `@deprecated` here.